website/docs/r/vapp_vm.html.markdown

```diff
@@ -343,9 +449,8 @@ The following arguments are supported:
 * `vdc` - (Optional; *v2.0+*) The name of VDC to use, optional if defined at provider level
 * `vapp_name` - (Required) The vApp this VM belongs to.
 * `name` - (Required) A name for the VM, unique within the vApp 
-* `computer_name` - (Optional; *v2.5+*) Computer name to assign to this virtual machine. 
-* `catalog_name` - (Optional; *v2.9+*) The catalog name in which to find the given vApp Template or media for `boot_image`.
-* `template_name` - (Optional; *v2.9+*) The name of the vApp Template to use
+* `computer_name` - (Optional; *v2.5+*) Computer name to assign to this virtual machine.
+* `template_id` - (Optional; *v3.8+*) The URN of the vApp Template to use. You can fetch it using a [`vcd_catalog_vapp_template`](/providers/vmware/vcd/latest/docs/data-sources/catalog_vapp_template) data source.
```

Comment:
CC: @dataclouder @Didainius

One reservation about naming this field: `template_id` or `vapp_template_id`?

Thoughts:
- `template_id` is more consistent with the previous `template_name` and is shorter to write
- `vapp_template_id` is more consistent with `vapp_name` and `catalog_vapp_template` data source, which makes it more intuitive without reading the docs

E.g. how it would look with `vapp_template_id`:

```hcl
data "vcd_catalog" "cat-where-is-template" {
  org  = "test"
  name = "cat-where-is-template"
}

data "vcd_catalog_vapp_template" "vappWithMultiVm" {
  org        = "test"
  catalog_id = data.vcd_catalog.cat-where-is-template.id
  name       = "vappWithMultiVm"
}

resource "vcd_vapp_vm" "secondVM" {
  vapp_name           = vcd_vapp.web.name
  vapp_template_id    = data.vcd_catalog_vapp_template.vappWithMultiVm.id
  vm_name_in_template = "secondVM" # Specifies which VM to use from the template
  name                = "secondVM"
  computer_name       = "db-vm"
  memory              = 512
  cpus                = 2
  cpu_cores           = 1
}
```

What do you think?

Reply:
We discussed that at some point also `vapp_name` will be changed to `vapp_id`, then `vapp_template_id` would be even more consistent:

```hcl
resource "vcd_vapp_vm" "secondVM" {
  vapp_id             = ...
  vapp_template_id    = data.vcd_catalog_vapp_template.vappWithMultiVm.id
  vm_name_in_template = "secondVM" # Specifies which VM to use from the template
  name                = "secondVM"
  computer_name       = "db-vm"
  memory              = 512
  cpus                = 2
  cpu_cores           = 1
}
```

Reply:
Changed to `vapp_template_id`!

Reply:
@adambarreiro Saw your refactoring commit, thanks!
Maybe change the four occurrences of `templateId` to `vappTemplateId` in code too?